onnxruntime/core/providers/snpe/snpe_runtime_options.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/snpe/snpe_runtime_options.h"
+#include "core/common/logging/logging.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace snpe {
+
+void SnpeRuntimeOptions::ParseOptions() {
+  static const std::string OPT_RUNTIME = "runtime";
+  static const std::string OPT_PRIORITY = "priority";
+  static const std::string BUFFER_TYPE = "buffer_type";
+
+  // Option - Runtime
+  if (const auto runtime_opt_it = runtime_options_.find(OPT_RUNTIME); runtime_opt_it != runtime_options_.end()) {
+    runtime_target_ = SnpeRuntimeWrapper(runtime_opt_it->second);
+    LOGS_DEFAULT(INFO) << "Located user specified runtime target: " << runtime_opt_it->second;
```

Comment:
If ort env is not created yet, here it would crash.  Like this: https://github.com/microsoft/onnxruntime/issues/8545